### PR TITLE
[stream] fixes for combined session failures, Errata UI

### PR DESCRIPTION
**Pull request by damoore044**
### Problem Statement
Initial PR #14773 raised by @LadislavVasina1, for 3 errata UI test fixes/updates.
- troubleshoot PRT failures in combined session, running the 3 tests from above PR.

Error in setup for `test_positive_check_errata_counts_by_type_on_host_details_page `:

Locally have intermittently reproduced issue with the `registered_contenthost` UI fixture with adding repos to cv, and promotion of cv to lce, in between parameters of combined session. Issues with some associated activation keys not being deleted in cleanup.

### PRT Case
```
trigger: test-robottelo
pytest: tests/foreman/ui/test_errata.py -k 'test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page or test_positive_list and not test_positive_list_permission'
```
- We run the 3 impacted tests from @LadislavVasina1 's PR and commits, 
   along with `test_positive_list`, which fails in CI and is fixed by the changes. 
   We exclude the otherwise collected `test_positive_list_permissions`, it is not impacted by these changes.

### CODE REVIEW

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_errata.py -k 'test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page'
airgun: 1337

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6580
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_errata.py -k test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page --external-logging
Test Result : ===== 2 passed, 13 deselected, 91 warnings, 1 error in 1154.36s (0:19:14) ======
```

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_errata.py -k 'test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page'
airgun: 1337

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6587
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_errata.py -k test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page --external-logging
Test Result : ===== 3 passed, 13 deselected, 97 warnings, 2 errors in 1314.73s (0:21:54) =====
```

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_errata.py -k 'test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page'
airgun: 1337

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_errata.py
airgun: 1337

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 113
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_errata.py -k test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page --external-logging
Test Result : ========= 3 passed, 13 deselected, 117 warnings in 1268.42s (0:21:08) ==========
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6605
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_errata.py -k test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page --external-logging
Test Result : ========= 3 passed, 13 deselected, 118 warnings in 1371.90s (0:22:51) ==========
```

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_errata.py -k 'test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 109
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_errata.py --external-logging
Test Result : ====== 8 failed, 7 passed, 1115 warnings, 5 errors in 5567.45s (1:32:47) =======
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6601
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_errata.py --external-logging
Test Result : ====== 8 failed, 6 passed, 1120 warnings, 8 errors in 5874.49s (1:37:54) =======
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6606
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_errata.py -k test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page --external-logging
Test Result : ========= 3 passed, 13 deselected, 121 warnings in 1347.37s (0:22:27) ==========
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 114
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_errata.py -k test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page --external-logging
Test Result : ========= 3 passed, 13 deselected, 122 warnings in 1480.89s (0:24:40) ==========
```

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_errata.py

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 117
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_errata.py --external-logging
Test Result : ======= 7 failed, 8 passed, 1189 warnings, 1 error in 5660.16s (1:34:20) =======
```

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6609
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/ui/test_errata.py --external-logging
Test Result : ======= 6 failed, 9 passed, 1194 warnings, 1 error in 5861.74s (1:37:41) =======
```

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_errata.py -k 'test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page or test_positive_list'

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_errata.py -k 'test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page or test_positive_list and not test_positive_list_permission'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6698
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_errata.py -k test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page or test_positive_list and not test_positive_list_permission --external-logging
Test Result : ========= 4 passed, 12 deselected, 169 warnings in 1394.12s (0:23:14) ==========
```

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_errata.py -k 'test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page or test_positive_list and not test_positive_list_permission'

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_errata.py -k 'test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page or test_positive_list and not test_positive_list_permission'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6702
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_errata.py -k test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page or test_positive_list and not test_positive_list_permission --external-logging
Test Result : ========= 4 passed, 12 deselected, 170 warnings in 1522.31s (0:25:22) ==========
```

**Review by vsedmik — APPROVED** [review]

**Review by vijaysawant — APPROVED** [review]

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_errata.py -k 'test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page or test_positive_list and not test_positive_list_permission'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6714
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_errata.py -k test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page or test_positive_list and not test_positive_list_permission --external-logging
Test Result : ========= 4 passed, 12 deselected, 167 warnings in 1510.77s (0:25:10) ==========
```

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_errata.py -k 'test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page or test_positive_list and not test_positive_list_permission'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6751
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_errata.py -k test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page or test_positive_list and not test_positive_list_permission --external-logging
Test Result : ========= 4 passed, 12 deselected, 166 warnings in 1417.19s (0:23:37) ==========
```

File: tests/foreman/ui/test_errata.py [hunk 1]
```diff
@@ -1081,112 +1090,99 @@ def test_positive_content_host_previous_env(
...
-    :Setup: Content Host with applicable errata
+    :id: f278f0e8-3b64-4dbf-a0c8-b9b289474a76
 
     :customerscenario: true
 
-    :steps: Search for errata on content host by type (e.g. 'type = security')
-     Step 1 Search for "type = security", assert expected amount and IDs found
-     Step 2 Search for "type = bugfix", assert expected amount and IDs found
-     Step 3 Search for "type = enhancement", assert expected amount and IDs found
+    :steps: Search for errata on the host Conetnt-Errata tab by type (e.g. 'type = Security')
```
Comment (Griffin-Sullivan) [inline tests/foreman/ui/test_errata.py:1138]:
```suggestion
    :steps: Search for errata on the host Content-Errata tab by type (e.g. 'type = Security')
```

File: tests/foreman/ui/test_errata.py [hunk 2]
```diff
@@ -1297,54 +1293,51 @@ def test_positive_show_count_on_host_pages(session, module_org, registered_conte
...
         assert vm.execute(f'yum install -y {pkgs}').status == 0
-
-        # navigate to content host main page by making a search, to refresh the details page
-        session.contenthost.search(hostname)
-        content_host_values = session.contenthost.read(hostname, 'details')
-        assert int(content_host_values['details']['security']) == FAKE_9_YUM_SECURITY_ERRATUM_COUNT
-
-        for errata_type in ('bug_fix', 'enhancement'):
-            assert int(content_host_values['details'][errata_type]) == 1
+        time.sleep(2)
```
Comment (Griffin-Sullivan) [inline tests/foreman/ui/test_errata.py:1334]: Why sleep here? If you're waiting for install it should already be done when you assert for status.

**Review by Griffin-Sullivan — APPROVED** [review]
One small fix and a question

File: tests/foreman/ui/test_errata.py (hunk 2 above)
Comment (damoore044) [reply tests/foreman/ui/test_errata.py:1334]: This was present from Ladislav's original commits. The package install is performed synchronously, waiting for status from result. The `sleep` here I believe is to wait for errata applicability recalculate task, performed after yum install. 

Locally I'm trying `wait_for_tasks` like we see in [`test_end_to_end()`](https://github.com/SatelliteQE/robottelo/blob/3f1b9e419c906275b739a03019c094960afed68f/tests/foreman/ui/test_errata.py#L458), thinking that should do the trick!

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_errata.py -k 'test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page or test_positive_list and not test_positive_list_permission'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6757
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_errata.py -k test_positive_check_errata or test_positive_errata_search_type or test_positive_check_errata_counts_by_type_on_host_details_page or test_positive_list and not test_positive_list_permission --external-logging
Test Result : ========= 4 passed, 12 deselected, 171 warnings in 1369.35s (0:22:49) ==========
```

**Review by sambible — APPROVED** [review]
ack pending griffin's comments.

**Pull request merged**